.github/workflows/github-actions-tests.yml

```diff
@@ -5,25 +5,22 @@ jobs:
   pre-commit:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-python@v4
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
         with:
           python-version: 3.11
-      - uses: pre-commit/action@v3.0.0
+      - uses: pre-commit/action@v3.0.1
   run-tests:
     strategy:
       fail-fast: false
       matrix:
-        python-version: [ "3.7", "3.8", "3.9", "3.10", "3.11" ]
-        mongodb-version: [ 4.4, 5.0 ]
-        pydantic-version: [ "1.10.15", "2.7" ]
-        exclude:
-          - python-version: "3.7"
-            pydantic-version: "2.7"
+        python-version: [ "3.8", "3.9", "3.10", "3.11","3.12"]
+        mongodb-version: [  5.0, 6.0, 7.0]
```

Comment:
We should still keep the MongoDB Version 4.4 unless we plan to explicitly remove support for that.

Reply:
will do